.github/workflows/run_nimpretty.yml

```
@@ -0,0 +1,37 @@
+---
+name: run_nimpretty
+
+# yamllint disable-line rule:truthy
+on:
+  workflow_dispatch:
+  push:
+
+jobs:
+  format_code_with_nimpretty:
+    name: format code with nimpretty
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - uses: jiro4989/setup-nim-action@v1
+      
+      - name: Run nimpretty
+        run: |
+          ./run_nimpretty.sh
+
+      - name: Configure git
+        run: |
+          git config --global user.name github-action
+          git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'
+
+      - name: Commit changes
+        run: |
+          if [[ `git status --porcelain` ]]; then
+            git commit -am "style: formatting with nimpretty"
+            git push origin HEAD:$GITHUB_REF
```

Creating and pushing a commit will not be allowed due to branch protection. Giving GitHub Actions full permissions is not recommended due to a few [number of reasons](https://github.com/orgs/community/discussions/25305#discussioncomment-3247401).
I recommend creating a PR and letting the maintainers review the PR created by the workflow, or we can merge it directly as well.

Also, I think it would be great if this could run on every `push` rather than workflow dispatch. 🙂

```suggestion
            git checkout -b directory-${{ github.sha }} # An unique branch name is recommended for each commit made on the repository.
            gh pr create --base ${GITHUB_REF##*/} --head directory-${{ github.sha }} --title 'style: formatting with `nimpretty`' --body '`Files have been modified with `nimpretty` (see diff. for changes)..'
```

---

Then maybe the workflow should simply _fail_, when the code if not formatted correctly. Having pull requests for almost every push might be impractical.

The suggested workflow can be triggered both manually (`workflow_dispatch:`) or by push.

---

> Then maybe the workflow should simply _fail_, when the code if not formatted correctly. Having pull requests for almost every push might be impractical. and automatic merging will also require write access.
> 
> The suggested workflow can be triggered both manually (`workflow_dispatch:`) or by push.

Write access in the repository settings shouldn't harm, and in the suggestion above I'm just creating a PR to the repository.
If you don't want on every push, you can just push changes to a certain branch, maybe. 🤔